### ISSUE TITLE
composer 2.0.13 is not supporting private repositories with local git 1.18.x

### ISSUE THREAD

**Issue by drekinov** [composer/composer#9863]
My `composer.json`:

```json
"repositories": [
    {
      "type": "vcs",
      "url": "ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE"
    }
  ],
```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 2.0.13
PHP version: 7.4.18
PHP binary path: /usr/bin/php
OpenSSL version: OpenSSL 1.0.2k  26 Jan 2017
cURL version: 7.29.0 libz 1.2.7 ssl NSS/3.53.1
zip: extension present, unzip present
```

When I run this command: 

```
composer update
```

I get the following output: 

```
Reading ./composer.json (/data/composer.json)
Loading config file ./composer.json (/data/composer.json)
Checked CA file /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem: valid
Executing command (/data): git branch -a --no-color --no-abbrev -v
Failed to initialize global composer: Composer could not find the config file: /root/.config/composer/composer.json

Reading /data/vendor/composer/installed.json
Loading plugin PackageVersions\Installer (from composer/package-versions-deprecated)
Running 2.0.13 (2021-04-27 13:11:08) with PHP 7.4.18 on Linux / 5.4.72-microsoft-standard-WSL2
Reading ./composer.lock (/data/composer.lock)
Loading composer repositories with package information
Executing command (/tmp): git remote -v
Executing command (/tmp): git ls-remote --heads -- 'ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE'
Executing command (CWD): git --version
Executing command (CWD): hg identify -- 'ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE'
Executing command (CWD): p4 -p 'ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE' info -s
Executing command (CWD): svn info --non-interactive -- 'ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE'


  [InvalidArgumentException]
  No driver found to handle VCS repository ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE


```

And I expected this to happen:
package to be updated from private repository.

=========

`git ls-remote --heads -- 'ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE` is not supported syntax on CentOS 7 with default git 1.18.x
what is working perfectly is `git ls-remote --heads 'ssh://git-codecommit.us-east-1.amazonaws.com/v1/repos/MYPACKAGEHERE`

basically i have to get git from thirdparty repository (End Point repository) as i can do that but i suppose that there are compliance situations where people are not allowed to install third party packages to their servers.

as soon as i reverted to composer 2.0.12 i was able to proceed with my work.
as CVE seems important probably i will have thirdparty git instead using fixed 2.0.12 while and if patch is even provided for 1.18.x

relevant change from 2.0.13 is  https://github.com/composer/composer/compare/2.0.12...2.0.13?s#diff-2f30cd17b349980f47ee8a67d55275661817b5a07f38395d7946140cf7660341

EDIT 2:
i figure out that `composer install` is working fine 2.0.13+git 1.18 but `composer update` fail.

**Comment by Seldaek** [comment]
IMO if this is only affecting the intersection of using git repos in their composer.json and having a super old git, this isn't really worth fixing. If you or someone who cares does the research to figure out exactly which git version added support for `--` in ls-remote though I'm open to fixing it, but in the meantime I'm closing this.
